website/src/views/pageStateHandlers/PageStateHandler.ts

```diff
@@ -16,10 +16,43 @@ export function toLapisFilterWithoutVariant(
     pageState: Dataset,
     constants: ExtendedConstants,
 ): LapisFilter & LapisLocation {
+    const baselineFilters = constants.baselineFilterConfigs?.reduce(
+        (acc, config) => {
+            switch (config.type) {
+                case 'date': {
+                    const value = pageState.datasetFilter[config.dateColumn] as DateRangeOption | undefined;
+                    if (value === undefined) {
+                        return acc;
+                    }
+
+                    return {
+                        ...acc,
+                        [`${config.dateColumn}From`]: value.dateFrom,
+                        [`${config.dateColumn}To`]: value.dateTo,
+                    };
+                }
+                case 'text': {
+                    const value = pageState.datasetFilter[config.lapisField] as string | undefined;
+                    if (value === undefined) {
+                        return acc;
+                    }
+
+                    return {
+                        ...acc,
+                        [config.lapisField]: value,
+                        [config.lapisField]: value,
+                    };
+                }
+            }
+        },
+        {} as { [key: string]: string | undefined },
+    );
```

Comment:
Another benefit of separating the state from the config:
* You can have an array of configs -> The filter fields will be displayed in the same order as in the array (maybe important, I don't know).
* You can have a more structured "state", separating the fields by type in an object like `{ textFilters: ..., dateFilters: ... }`. That would turn this huge switch-case into two loops.